annotation/tests/test_annotations.py

```diff
@@ -235,3 +274,156 @@ def test_create_manifest_json(
             annotated_doc.tenant,
             moto_s3,
         )
+
+
+@pytest.mark.parametrize(
+    ("is_latest", "new_equal_latest"),
+    ((True, False), (True, True)),  # Success path  # return latest_doc
+)
+def test_construct_annotated_doc(
+    annotated_doc,
+    annotation_doc_for_save,
+    is_latest,
+    new_equal_latest,
+):
+    doc = annotation_doc_for_save
+    latest_doc = annotated_doc
+    s3_path = (
+        f"{S3_START_PATH}/{annotated_doc.job_id}/" f"{annotated_doc.file_id}"
+    )
+    s3_file_path = "path"
+    s3_file_bucket = "bucket"
+
+    pages_sha_latest = ({"1": "a"}, "b")
+    pages_sha_non_latest = ({"1": "a"}, "c")
+
+    db = Mock()
+
+    with patch.object(db, "add") as mock_db_add, patch.object(
+        db, "add_all"
+    ) as mock_db_add_all, patch.object(
+        db,
+        "commit",
+    ) as mock_db_commit, patch(
+        "annotation.annotations.main.get_pages_sha",
+        return_value=pages_sha_latest if is_latest else pages_sha_non_latest,
+    ) as mock_get_pages_sha, patch(
+        "annotation.annotations.main.check_docs_identity",
+        return_value=new_equal_latest,
+    ) as mock_check_docs, patch(
+        "annotation.annotations.main.construct_document_links", return_value=[]
+    ) as mock_construct_doc_links, patch(
+        "annotation.annotations.main.convert_bucket_name_if_s3prefix",
+        return_value=TEST_TENANT,
+    ) as mock_convert_bucket, patch(
+        "annotation.annotations.main.upload_pages_to_minio"
+    ) as mock_upload_pages, patch(
+        "annotation.annotations.main.create_manifest_json"
+    ) as mock_create_manifest:
+        construct_annotated_doc(
+            db,
+            annotated_doc.user,
+            None,
+            annotated_doc.job_id,
+            annotated_doc.file_id,
+            doc,
+            TEST_TENANT,
+            s3_file_path,
+            s3_file_bucket,
+            latest_doc,
+            1,  # task_id
+            is_latest,
+        )
+        if is_latest:
+            mock_get_pages_sha.assert_called_once_with(
+                doc.pages,
+                doc.base_revision,
+                doc.validated,
+                doc.failed_validation_pages,
+                doc.user,
+            )
+        else:
+            mock_get_pages_sha.assert_called_once_with(
+                doc.pages,
+                latest_doc.revision,
+                doc.validated,
+                doc.failed_validation_pages,
+                doc.user,
```

Comment:
Remove logic from the tests. Conditional if statements are often the most common source of errors in tests. It might be beneficial to split this function into several parts to improve clarity and reduce potential issues.